apps/choleskey/cholesky_tiled.cpp

```diff
@@ -0,0 +1,161 @@
+
+    // Check mat_size is divisible by num_tiles
+    if (matrix_size % num_tiles != 0) {
+        fmt::print("matrix size must be divisible by num_tiles.. aborting\n");
+        throw std::invalid_argument("Matrix size is not divisible by num_tiles");
+    }
+
+    if (matrix_size == 0) {
+        fmt::print(" 0 illegal input matrix_size \n");
+        std::exit(0);
+    }
+
+    tile_size = matrix_size / num_tiles;
+    tot_tiles = num_tiles * num_tiles;
+
+    data_type* A = new data_type[matrix_size * matrix_size];
+
+    // Allocate memory for tiled_cholesky for the full matrix
+    data_type* A_lower = new data_type[matrix_size * matrix_size];
+
+    // Allocate memory for MKL cholesky for the full matrix
+    data_type* A_MKL = new data_type[matrix_size * matrix_size];
+
+    // Memory allocation for tiled matrix
+    data_type** Asplit = new data_type*[tot_tiles];
+
+    for (int i = 0; i < tot_tiles; ++i) {
+        // Buffer per tile
+        Asplit[i] = new data_type[tile_size * tile_size];
+    }
+
+    //Generate a symmetric positve-definite matrix
+    A = generate_positiveDefinitionMatrix(matrix_size);
+
+    CBLAS_ORDER blasLay;
+    int lapackLay;
+
+    if (layRow) {
+        blasLay = CblasRowMajor;
+        lapackLay = LAPACK_ROW_MAJOR;
+    } else {
+        blasLay = CblasColMajor;
+        lapackLay = LAPACK_COL_MAJOR;
+    }
+
+    //copying matrices into separate variables for tiled cholesky (A_lower) and MKL cholesky (A_MKL)
+    std::memcpy(A_lower, A, matrix_size * matrix_size * sizeof(data_type));
+    std::memcpy(A_MKL, A, matrix_size * matrix_size * sizeof(data_type));
+
+    //splits the input matrix into tiles
+    split_into_tiles(A_lower, Asplit, num_tiles, tile_size, matrix_size, layRow);
+
+    // Measure execution time.
+    Timer timer;
+    //run the tiled Cholesky function
+    tiled_cholesky(Asplit, tile_size, num_tiles, blasLay, lapackLay);
+
+    //assembling seperated tiles back into full matrix
+    assemble_tiles(Asplit, A_lower, num_tiles, tile_size, matrix_size, layRow);
+    auto time = timer.stop();
+
+    if (args.time) {
+        fmt::print("Duration: {:f} ms\n", time);
+    }
+
+    //calling LAPACKE_dpotrf cholesky for verification and timing comparison
+    int info = LAPACKE_dpotrf(lapackLay, 'L', matrix_size, A_MKL, matrix_size);
+
+    if (verifycorrectness == 1) {
+        bool res = verify_results(A_lower, A_MKL, matrix_size * matrix_size);
+        if (res) {
+            fmt::print("Tiled Cholesky decomposition successful\n");
+        } else {
+            fmt::print("Tiled Cholesky decomposition failed\n");
+        }
+        fmt::print("\n");
+    }
+
+    // print lower_matrix if tiled_cholesky sucessfull
+    if (args.lower_matrix) {
+        fmt::print("The lower matrix of input matrix after tiled_cholesky: \n");
+        printLowerResults(A_lower, matrix_size);
+    }
+
+    //memory free
+    delete[] A;
+    delete[] A_lower;
+    delete[] A_MKL;
```

Comment:
nit, I wonder if we can do it thorough unique_ptr instead of new and delete. not important for this PR though.